src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/configuration/TemplateLoader.java

```diff
@@ -65,12 +65,6 @@ public RootBuilder getTemplate(FeatureTypeInfo typeInfo, String outputFormat)
         if (template.checkTemplate()) updateCache = true;
 
         RootBuilder root = template.getRootBuilder();
-        // check if reload is needed anyway and eventually reload the template
```

Comment:
Not sure why this is gone?

Reply:
The reload now happens entirely in the checkTemplate method called above (calling RootBuilder#needsReload().
There was no reason to keep that piece of code calling the needsReload method from the TemplateLoader.